pkg/kubelet/metrics/collectors/cri_metrics.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
```

Comment:
I think we need to move this to `k8s.io/component-base/metrics` based on discussion I had with @dashpole. 

Can we use that + `NewLazyConstMetric` - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/metrics/value.go#L43? Or is something preventing us?

Reply:
the problem is if the CRI implementation messes up the metric, it causes kubelet to panic. I tried that originally, but in my fussing with the metrics, I kept causing kubelet to crash. NewLazyConstMetric seems to be more in situations we have the metrics/descriptions precalculated and are willing to panic if they're wrong (programmer error vs runtime error)

Reply:
That is pretty concerning...

If we need it here, we would need to add NewConstMetric (without Must) to component-base, rather than use the upstream prometheus library.  cc @logicalhan

Reply:
The other reason I went for this is there isn't really a notion of stability with these metrics (unless we wire them through the cri, or assume they have the stability of the KEP)

would you prefer I:
- add NewConstMetric function
- use MustNewConstMetric despite the risk
- continue this way until we have a more clear path forward?

From my perspective, cadvisor is currently registered with RawMustRegister, so this way is the most in-line with cadvisor's behavior. I feel like it could be a follow-up to update

Reply:
I would prefer:

* add NewConstMetric function

We should either:

* Mark all metrics as Internal stability, or
* Add stability to the CRI API, and use that.